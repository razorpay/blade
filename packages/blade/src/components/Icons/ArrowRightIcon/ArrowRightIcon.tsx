import { Svg, Path } from '../_Svg';
import type { IconComponent } from '..';
import useIconProps from '../useIconProps';

const ArrowRightIcon: IconComponent = ({ size, color, ...styledProps }) => {
  const { height, width, iconColor } = useIconProps({ size, color });

  return (
    <Svg {...styledProps} width={width} height={height} viewBox="0 0 24 24" fill="none">
      <Path
        d="M14.7071 5.29289C14.3166 4.90237 13.6834 4.90237 13.2929 5.29289C12.9024 5.68342 12.9024 6.31658 13.2929 6.70711L17.5858 11H4C3.44772 11 3 11.4477 3 12C3 12.5523 3.44772 13 4 13H17.5858L13.2929 17.2929C12.9024 17.6834 12.9024 18.3166 13.2929 18.7071C13.6834 19.0976 14.3166 19.0976 14.7071 18.7071L20.7071 12.7071C21.0976 12.3166 21.0976 11.6834 20.7071 11.2929L14.7071 5.29289Z"
        fill={iconColor}
      />
    </Svg>
  );
};

export default ArrowRightIcon;
