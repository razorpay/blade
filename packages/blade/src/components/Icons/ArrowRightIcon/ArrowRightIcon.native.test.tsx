import ArrowRightIcon from '.';
import renderWithTheme from '~src/_helpers/testing/renderWithTheme.native';

describe('<ArrowRightIcon />', () => {
  it('should render ArrowRightIcon', () => {
    const renderTree = renderWithTheme(
      <ArrowRightIcon color="feedback.icon.neutral.lowContrast" size="large" />,
    ).toJSON();
    expect(renderTree).toMatchSnapshot();
  });
});
