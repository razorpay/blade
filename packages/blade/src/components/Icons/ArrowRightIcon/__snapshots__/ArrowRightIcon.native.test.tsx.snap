// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ArrowRightIcon /> should render ArrowRightIcon 1`] = `
<View
  style={
    {
      "flex": 1,
    }
  }
>
  <RNSVGSvgView
    accessibilityElementsHidden={true}
    align="xMidYMid"
    bbHeight="20px"
    bbWidth="20px"
    data-blade-component="icon"
    fill="none"
    focusable={false}
    height="20px"
    importantForAccessibility="no-hide-descendants"
    meetOrSlice={0}
    minX={0}
    minY={0}
    style={
      [
        {
          "backgroundColor": "transparent",
          "borderWidth": 0,
        },
        [
          {},
        ],
        {
          "flex": 0,
          "height": 20,
          "width": 20,
        },
      ]
    }
    vbHeight={24}
    vbWidth={24}
    width="20px"
  >
    <RNSVGGroup
      fill={null}
      propList={
        [
          "fill",
        ]
      }
    >
      <RNSVGPath
        d="M14.707 5.293a1 1 0 1 0-1.414 1.414L17.586 11H4a1 1 0 1 0 0 2h13.586l-4.293 4.293a1 1 0 0 0 1.414 1.414l6-6a1 1 0 0 0 0-1.414l-6-6Z"
        fill={
          {
            "payload": 4280563015,
            "type": 0,
          }
        }
        propList={
          [
            "fill",
          ]
        }
      />
    </RNSVGGroup>
  </RNSVGSvgView>
</View>
`;
